src/api/app/controllers/status_project_controller.rb

```diff
@@ -1,12 +1,21 @@
 class StatusProjectController < ApplicationController
+  before_action :initialize_caches, only: [:show]
+
   def show
     dbproj = Project.get_by_name(params[:project])
     @packages = ProjectStatus::Calculator.new(dbproj).calc_status
-    find_relationships_for_packages(@packages)
+    find_relationships_for_packages
   end
 
   private
 
+  def initialize_caches
```

Comment:
I think the more pressing question is. Why do we cache this?

Reply:
memoizing maybe helps in the ```each`` block?

Reply:
maybe?

Reply:
:musical_note: Hey, I just wrote it and this is craaaazy
But here's my PoC, so call me maaaybe  :musical_note: